### PR TITLE
Replace peekalink with OpenGraph Ninja API

### PR DESCRIPTION
**Pull request by SSHari**
PR to switch from Peekalink to OpenGraph Ninja. Switching to the OpenGraph Ninja API was fairly straightforward, but there are a few differences that should be called out:

1. OpenGraph provides `alt` tag information, so I added that to the `PreviewHtml` component
2. OpenGraph doesn't provide an icon image in the response, but it does return a website's `favicon` image. I switched the `icon` image to point to this instead
3. OpenGraph doesn't provide additional information for Twitter or Youtube (e.g. `likeCount`). Without this data available, it didn't seem to make sense to keep the `PreviewProperties` logic in the `PreviewHtml` component, so I removed it.

It does look like in _some_ cases at least, OpenGraph is doing better than Peekalink. I tried to preview [an API Hero blog post](https://blog.apihero.run/api-hero-an-open-source-proxy-that-makes-every-api-you-use-faster-and-more-reliable) in the currently deployed version of the app and it couldn't generate a preview: https://jsonhero.io/j/zGYmkaZxqyyf.

image

In my local version of the app which is using OpenGraph, it does seem to load the preview correctly:

image

I think whether or not we want to move forward with this change depends on how important point 3 is above. Given that [OpenGraph is free](https://opengraph.ninja/pricing), this tradeoff might be acceptable, but ultimately the maintainers of JSON Hero will need to decide.

Bounty: /claim #10

### CODE REVIEW

**Comment by ericallam** [comment]
This looks great! Regarding point 3 above, it's fine if we lose the PreviewProperties since OpenGraph doesn't support it. I'll review this PR shortly and leave any comments in the review 👍

File: app/services/uriPreview.server.ts [hunk 1]
```diff
@@ -71,9 +77,7 @@ export async function getUriPreview(uri: string): Promise<PreviewResult> {
     return createPreviewJson(url.href, jsonBody);
   }
 
-  const peekalinkResult = await getPeekalink(url.href);
-
-  return peekalinkResult;
+  return await getOpenGraphNinja(`${url.hostname}${url.pathname}${url.search}`);
```
Comment (ericallam) [inline app/services/uriPreview.server.ts:80]: Is there a reason this isn't using `url.href` like the previous code was?

**Review by ericallam — CHANGES_REQUESTED** [review]
This looks great, could you double check the one comment I left and then after that should be good to merge!

File: app/services/uriPreview.server.ts (hunk 1 above)
Comment (SSHari) [reply app/services/uriPreview.server.ts]: @ericallam This was a mistake on my part. I misread the documentation for OpenGraph. I thought it said we needed to use a URL _without_ the protocol, but it's saying you can use _either_. Good catch! I've gone ahead and switched it back to using the `url.href` 👍.

image

**Review by ericallam — APPROVED** [review]

**Pull request merged**